cmake/ccache.cmake

```diff
@@ -8,45 +8,57 @@ if (CMAKE_CXX_COMPILER_LAUNCHER MATCHES "ccache" OR CMAKE_C_COMPILER_LAUNCHER MA
     return()
 endif()
 
-option(ENABLE_CCACHE "Speedup re-compilations using ccache (external tool)" ON)
+option(ENABLE_CCACHE "Speedup re-compilations using ccache (default) or sccache (external tools)" ON)
+option(FORCE_SCCACHE "Use sccache by default" OFF)
 
-if (NOT ENABLE_CCACHE)
-    message(STATUS "Using ccache: no (disabled via configuration)")
+if (NOT ENABLE_CCACHE AND NOT FORCE_SCCACHE)
+    message(STATUS "Using *ccache: no (disabled via configuration)")
     return()
 endif()
 
-find_program (CCACHE_EXECUTABLE ccache)
+# We support ccache and sccache
+if (FORCE_SCCACHE)
+    find_program (CCACHE_EXECUTABLE sccache)
+else()
+    find_program (CCACHE_EXECUTABLE ccache sccache)
```

Comment:
I am confused, should there really be a fallback from ccache to sccache? The call will be through `CACHE_EXECUTABLE`, are both binaries drop-replacements (exchangeable)? Maybe the logic should be less implicit (see my above comment).

Reply:
Well, we are not operating a nuclear power plant here 😄  The only effect would be unexpectedly slow builds. And CMake will print a warning if an option (`-D...`) was passed but not used, so users will notice.

Anyways, a cleaner separation of ccache  and sccache would be nice but it's up to you.

Reply:
It's addressed in https://github.com/ClickHouse/ClickHouse/commit/8efa2ce84390eaa26e8c718660273de08032e98a